src/Pagination/components/PaginationContainer.ts

```diff
@@ -236,27 +238,45 @@ export default class PaginationContainer extends Component<ModelerProps, Paginat
                 showLoader(targetListView);
 
                 this.setState({
+                    pageSize: pageSize || this.state.pageSize,
                     currentOffset: offSet,
                     currentPageNumber: pageNumber,
                     isLoadingItems: true
                 });
 
                 targetListView._datasource.setOffset(offSet);
                 targetListView.sequence([ "_sourceReload", "_renderData" ]);
-                this.publishListViewUpdate(offSet, pageNumber);
+                this.publishListViewUpdate({ newOffSet: offSet, newPageNumber: pageNumber });
             }
         }
     }
 
-    private publishListViewUpdate(offSet: number, pageNumber: number) {
+    private publishListViewUpdate = (topicMessage: TopicMessage) => {
         if (this.state.targetListView) {
-            dojoTopic.publish(this.subscriptionTopic, [ offSet, pageNumber ]);
+            dojoTopic.publish(this.subscriptionTopic, topicMessage);
         }
     }
 
     private isHideUnUsed(targetListView: ListView): boolean {
-        const offset = targetListView._datasource._pageSize;
+        const pageSize = targetListView._datasource._pageSize;
 
-        return ((offset >= targetListView._datasource._setSize) || (offset === 0)) && this.props.hideUnusedPaging;
+        return ((pageSize >= targetListView._datasource._setSize) || (pageSize === 0)) && this.props.hideUnusedPaging;
+    }
+
+    private applyPageSize = (onChangeProps: OnChangeProps) => {
+        const { newPageSize, newOffSet } = onChangeProps;
+        const { targetListView } = this.state;
+
+        if (targetListView && targetListView._datasource
+                && targetListView._datasource._pageSize !== newPageSize) {
```

Comment:
Seems like it can survive just fine on the above line...have seen longer lines in your code hehe

Reply:
Thanks, its perfect here :+1: